### ISSUE TITLE
[Request]: Create docs for new user details flyout UI

### ISSUE THREAD

**Issue by machadoum** [elastic/security-docs#4459]
### Description

We need documentation for the new user details flyout UI.

Why? The previous version of the flyout was restrictive. It prevented us from quickly adding new features.
The new user flyout version is aligned with the latest alerts flyout UI.

### Related links / assets

Please include each of the following, if applicable:
Subject matter expert: @SourinPaul 
Figma link(s): [Initial design](https://www.figma.com/file/fvWYDmiRM7E6UokAYzLleu/Design-%7C-Entity-Flyout-%7C-Entity-Analytics?type=design&node-id=664-245516&mode=design&t=ekhHq4cz3kcXXGiS-0) (It evolved after interactions)
Github epic link(s): https://github.com/elastic/security-team/issues/5837
Github issue link(s): https://github.com/elastic/security-team/issues/8007 https://github.com/elastic/security-team/issues/8038
Dev docs for managed data section: https://docs.google.com/document/d/1e51KwcqtnhELz7k_xTFAYIUZ6EsrfUAObdJMGpCVjF0
docs issue for the host flyout: https://github.com/elastic/security-docs/issues/4576
### Which documentation set needs improvement?

ESS and serverless

### Software version

The feature was introduced on version 8.12.0 behind an experimental flag.

### Collaborators

PM: @SourinPaul 
Designer: Pavel Manko
Developer: @machadoum 
Others (if applicable): @jared

### Timeline / deliverables

TBD @SourinPaul @jaredburgettelastic Can you help me here?

**Comment by SourinPaul** [comment]
Thanks, @machadoum.

Team, I want to revisit our question: `Do we need to document a feature that is behind a Feature Flag?` 

Reasoning: We have decided not to remove FF in 8.12. Furthermore, we decided to park this feature in favor of higher priorities in 8.13. We expect to resume in 8.14, complete pending flyout updates, and remove the FF before releasing. 

With this, what are ourr expectations for releasing user-facing documentation with interim updates from 8.12? Should we release docs for the current state or wait until the feature is complete? 
@natasha-moore-elastic @janeen kindly advise. Thanks!
